src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -13,6 +13,14 @@ namespace System.Net.Sockets
 {
     internal sealed unsafe class SocketAsyncEngine
     {
+        // Socket continuations are dispatched to the ThreadPool from the event thread.
+        // This avoids continuations blocking the event handling.
+        // Setting DOTNET_SYSTEM_NET_SOCKETS_INLINE_CONTINUATIONS to '1'
+        // runs continuations directly on the event thread.
+        // This removes the overhead of context switching to the ThreadPool,
+        // at the risk of potentially blocking the event thread.
+        internal static readonly bool InlineSocketContinuations = Environment.GetEnvironmentVariable("DOTNET_SYSTEM_NET_SOCKETS_INLINE_CONTINUATIONS") == "1";
```

Comment:
Thinking more about this, we should probably split this into two env vars: one that controls the number of epoll threads, and one that controls socket continuations.  We probably want the former for easier experimentation as well as if/when we have an API on Socket that controls inline continuations, so that it can be controlled separately from the latter.  The latter would then solely be about configuring the default for the API on Socket.

Reply:
Makes sense. We've been wanting to control the thread count during our benchmarks.
I've added an envvar. ptal.